File: domains/cfsr/scripts/build_presentation_template.R
# build_presentation_template.R — generate blank officer-compatible .pptx templates
#
# Purpose: Write states/{ky,md}/_assets/*-presentation-template.pptx (16:9, default layouts).
# Inputs: None (requires Python 3 + python-pptx).
# Outputs: ky-presentation-template.pptx, md-presentation-template.pptx
#
# Run from monorepo root:
#   Rscript domains/cfsr/scripts/build_presentation_template.R

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)
script_dir <- if (length(file_arg)) {
  dirname(sub("^--file=", "", file_arg[1]))
} else {
  getwd()
}
root <- normalizePath(file.path(script_dir, "..", "..", ".."), winslash = "/", mustWork = TRUE)
setwd(root)

py_script <- file.path(root, "domains", "cfsr", "scripts", "build_presentation_template.py")
if (!file.exists(py_script)) {
  stop("Missing: ", py_script)
}

py <- Sys.which("python")
if (py == "") py <- Sys.which("python3")
if (py == "") stop("Python not found on PATH; install Python 3 to build templates.")

rc <- system2(py, args = py_script, wait = TRUE, stdout = FALSE, stderr = FALSE)
if (rc != 0) {
  stop("build_presentation_template.py failed with exit code ", rc)
}

message("Templates OK.")

File: domains/cfsr/scripts/build_presentation_template.py
"""
Copy the branded CFSR presentation template to state-specific locations.

Kurt's branded template (cfsr-presentation-template.pptx) contains custom layouts:
- Title Slide
- Title and Content
- Section Header
- Side Panel with Picture (for indicator slides)
- Top Banner with Picture (for summary app)

Run from repo root:
  python domains/cfsr/scripts/build_presentation_template.py
"""

from __future__ import annotations

import shutil
import sys
from pathlib import Path


def main() -> int:
    root = Path(__file__).resolve().parents[3]
    
    # Source: Kurt's branded template
    source_template = root / "domains" / "cfsr" / "templates" / "cfsr-presentation-template.pptx"
    
    if not source_template.exists():
        print(f"ERROR: Branded template not found at {source_template}")
        print("Please ensure cfsr-presentation-template.pptx exists in domains/cfsr/templates/")
        return 1
    
    # Destinations: state-specific template locations
    out_ky = root / "states" / "ky" / "_assets" / "ky-presentation-template.pptx"
    out_md = root / "states" / "md" / "_assets" / "md-presentation-template.pptx"
    
    out_ky.parent.mkdir(parents=True, exist_ok=True)
    out_md.parent.mkdir(parents=True, exist_ok=True)

    # Copy the branded template to state locations
    shutil.copy2(source_template, out_ky)
    shutil.copy2(source_template, out_md)
    
    print(f"Copied branded template to {out_ky}")
    print(f"Copied branded template to {out_md}")
    return 0


if __name__ == "__main__":
    sys.exit(main())
